src/components/MarkdownProvider/components/BestPractice.tsx

```diff
@@ -118,6 +123,7 @@ interface ISectionProps extends ICaptionProps {
   imageHeight?: number;
   imageWidth?: number;
   imageIsSquare?: boolean;
+  imageBackgroundColor?: string;
```

Comment:
The image should be sized accordingly to fill in the space. I think design should update the image so that it fits correctly.

Note on props: I think adding new props should be considered carefully. If a new prop is added for each new requirement, it could lead to components having dozens of props with take away rendering control from the consumer. IMO, props should only be added if its critical part of component implementation — otherwise build the component such that consumers can compose behavior + style.

Reply:
I do agree, we should remove the provided padding on our end to allow designers to have more control of the examples. I'll remove the extra prop and leave `Don't use 32px icons for UI` as to how it was before.

I'm guessing this can be potentially a separate PR to address the padding removal from the examples?

For context:
One of the examples had a filled background for the image which looks a bit odd due to the padding.
<img width="791" alt="image" src="https://user-images.githubusercontent.com/30312964/154324122-5dad92e2-9caf-48c4-befb-ac23ef039bc5.png">

With the `imageBackgrounfColor`
<img width="792" alt="image" src="https://user-images.githubusercontent.com/30312964/154324289-de5e36c0-09c3-4a5d-8f84-ac09a712d294.png">

Reply:
Oh I see, the padding is an issue so the image can't be re-sized to fit to the edges of the border. Seems like that's the intentional design of the component: image fits within the padding. Coloring the background gray to make the padding appear as part of the image is breaking that intention. Plus, the API for achieving that isn't great.

Personally I would a) abide by the current intention and leave the white spaced padding showing or b) I would adjust the image container to have zero padding and all images fit the container. Option b) is a separate task that needs team discussion.

Reply:
I asked him to do this. Is there a reason why changing the background color is bad? I think it looks better and I assume it isn't too disruptive to the structure of the page.

Reply:
I agree I think it looks better. I think changing the background color to achieve it is bad because the `Dont` component was built to have padding around images. It looks like the reason for having padding was possibly flawed, so that should be looked into. Also, changing the background color could mislead the user when:

1) A user attempts to save an image. They right click the background color to save the image expecting the ability to right-click-and-save, it won't work since the clicked area is not part of the image.

2) A user saves the image. The saved image does not reflect what they saw on the website. The background color is much less in the image they saved.

I assume website visitors have no intention to save images, but these issue two points reveal flaws with the approach of changing the background color to achieve what the image should originally look like. The background color approach could be a temporary fix and there could be a separate task to investigate this issue.

Reply:
Hmm I may be missing something here... where is `imageBackgroundColor` actually implemented?